src/main/java/org/jenkinsci/plugins/pipeline/utility/steps/zip/ZipStepExecution.java

```diff
@@ -93,6 +93,14 @@ protected Void run() throws Exception {
         int count = source.act(new ZipItFileCallable(destination, step.getGlob()));
         listener.getLogger().println("Zipped " + count + " entries.");
         if (step.isArchive()) {
+            Run build = getContext().get(Run.class);
+            if (build == null) {
+                throw new MissingContextVariableException(Run.class);
```

Comment:
unnecessary, just declare in required context

Reply:
The `Run.class` is only required if `step.isArchive()` is `true`

Reply:
or do you mean that there should always be a `Run.class` available anyway, so no need for the hassle?

Reply:
@rsandell should always implicitly have the Run available, I think

Reply:
No. You must declare it in `getRequiredContext`. Then you can obtain it without a null check.